### PR TITLE
Add Wingman and caching to CI workflow

### PR DESCRIPTION
**Pull request by ligaz**
This change improves our CI workflow by adding node modules caching and Wingman. 

Let the dogfooding begin 🐶 💩

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -20,9 +23,14 @@ jobs:
         uses: actions/setup-node@v4
         with:
           node-version-file: ".nvmrc"
+          cache: "npm"
 
       - name: Install dependencies
-        run: npm install
+        run: npm ci
```
Comment (harveysmurf) [inline .github/workflows/ci.yml:29]: why is this changed?

Comment (ligaz) [reply .github/workflows/ci.yml:29]: https://letmegooglethat.com/?q=npm+install+vs+npm+ci

**Review by harveysmurf — APPROVED** [review]
